File: TourDeFrance.Ui.Angular/models/drink.ts
﻿class Drink {
    id: string;
    name: string;
    ownerId: string;
    alcoholByVolume: number;
    volume: number;
    subDrinks: SubDrink[];
    isComposedDrink: boolean;
    calculatedVolume: number;
    calculatedAlcoholByVolume: number;
}

class SubDrink {
    id: string;
    drinkId: string;
    name: string;
    alcoholByVolume: number;
}

File: TourDeFrance.Client/Drink/Drink.cs
using System;

namespace TourDeFrance.Client.Drink
{
	public class Drink
	{
		public Guid Id { get; set; }

		public string Name { get; set; }

		public Guid OwnerId { get; set; }

		// NOTE: in percent
		public decimal? AlcoholByVolume { get; set; }

		// NOTE: in centiliter
		public decimal? Volume { get; set; }

		public SubDrink[] SubDrinks { get; set; }

		public bool IsComposedDrink { get; set; }

		public decimal CalculatedVolume { get; set; }

		public decimal CalculatedAlcoholByVolume { get; set; }
	}
}
